src/root/views/Country/ThreeW/InCountryProjects/Map/index.tsx

```diff
@@ -0,0 +1,244 @@
+  countryId?: number;
+  projectList: Project[];
+}
+
+function ThreeWMap(props: Props) {
+  const {
+    className,
+    projectList,
+    countryId,
+  } = props;
+
+  const allCountries = useReduxState('allCountries');
+  const countryBounds = React.useMemo(() => (
+    turfBbox(allCountries?.data.results.find(
+      d => d.id === countryId)?.bbox ?? []
+    )
+  ), [allCountries, countryId]);
+
+  const {
+    response: districtListResponse,
+  } = useRequest<ListResponse<District>>({
+    skip: !countryId,
+    url: 'api/v2/district/',
+    query: {
+      country: countryId,
+      limit: 200,
+    },
+  });
+
+  const districtDenormalizedProjectList = React.useMemo(
+    () => denormalizeList(
+      projectList ?? [],
+      (p) => p.project_districts_detail,
+      (p, d) => ({
+        ...p,
+        project_district: d,
+      }),
+    ),
+    [projectList]
+  );
+
+  const districtGroupedProjects = listToGroupList(
+    districtDenormalizedProjectList,
+    d => d.project_district.id,
+  );
+
+  const districtList = districtListResponse?.results ?? emptyDistrictList;
+
+  const [
+    clickedPointProperties,
+    setClickedPointProperties,
+  ] = React.useState<ClickedPoint| undefined>();
+
+  const geo: ProjectGeoJson = React.useMemo(
+    () => ({
+      type: 'FeatureCollection' as const,
+      features: districtList.map((district) => {
+        const projects = districtGroupedProjects[district.id];
+        if (!projects) {
+          return undefined;
+        }
+
+        return {
+          id: district.id,
+          type: 'Feature' as const,
+          properties: {
+            districtId: district.id,
+            numProjects: projects.length,
+          },
+          geometry: {
+            type: 'Point' as const,
+            coordinates: district.centroid?.coordinates ?? [0, 0],
+          },
+        };
+      }).filter(isDefined),
+    }),
+    [districtList, districtGroupedProjects],
+  );
+
+  const selectedDistrictProjectDetail = React.useMemo(
+    () => {
+      if (!clickedPointProperties || !clickedPointProperties.feature?.id) {
+        return undefined;
+      }
+
+      const selectedDistrictProjectList = districtGroupedProjects[clickedPointProperties.feature.id];
+
+      if (!selectedDistrictProjectList) {
+        return undefined;
+      }
+
+      return selectedDistrictProjectList;
+    },
+    [clickedPointProperties, districtGroupedProjects],
+  );
+
+  const handlePointClick = React.useCallback(
+    (feature: mapboxgl.MapboxGeoJSONFeature, lngLat: mapboxgl.LngLat) => {
+      setClickedPointProperties({
+        feature: feature as unknown as ClickedPoint['feature'],
+        lngLat,
+      });
+      return true;
+    },
+    [setClickedPointProperties],
+  );
+
+  const handlePointClose = React.useCallback(
+    () => {
+      setClickedPointProperties(undefined);
+    },
+    [setClickedPointProperties],
+  );
+
+  const maxScaleValue = Math.max(projectList?.length ?? 0, 2);
```

Comment:
We may need to add comment why we did Math.max with value 2

Reply:
Maybe we should move this Math.max inside the helper utility where the value `2` makes sense